providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/EventStreamObserver.java

```diff
@@ -60,23 +61,13 @@ public void onCompleted() {
         this.callback.setEventStreamAlive(false);
     }
 
-    private void handleConfigurationChangeEvent(EventStreamResponse value) {
+    private void handleConfigurationChangeEvent() {
+        this.callback.emitConfigurationChangeEvent();
         if (!this.cache.getEnabled()) {
             return;
         }
-
-        Map<String, Value> data = value.getData().getFieldsMap();
-        Value flagsValue = data.get(flagsKey);
-        if (flagsValue == null) {
-            this.cache.clear();
-            return;
-        }
-
-        Map<String, Value> flags = flagsValue.getStructValue().getFieldsMap();
-
-        for (String flagKey : flags.keySet()) {
-            this.cache.remove(flagKey);
-        }
+        // Always flush the cache when configuration_change event is received
+        this.cache.clear();
```

Comment:
@toddbaert could you please double-check that I interpreted [the docs](https://github.com/open-feature/flagd/blob/main/docs/other_resources/creating_providers.md#provider-lifecycle-initialization-and-shutdown) correctly: 

> emit PROVIDER_CONFIGURATION_CHANGED event and flush cache when a configuration_change message is received on the streaming connection

Reply:
No, I think because we know the flags changed, we don't need to flush them all (this part hasn't changed). I think I should update the doc to be more precise about this.

In the case of flagd, we know exactly the flags changed, so we can be more efficient by only busting those keys.

Reply:
Thanks! I have updated the PR and tests :) now it is ready for reviews